cpu/msp430fxyz/periph/spi.c

```diff
@@ -210,18 +211,40 @@ int spi_release(spi_t dev)
     return 0;
 }
 
-int spi_transfer_byte(spi_t dev, char out, char *in)
+int spi_transfer_bytes(spi_t dev, char *out, char *in, unsigned int length)
 {
     (void)dev;
-    char tmp;
-    while (!(SPI_IF & SPI_IE_TX_BIT)) {}
-    SPI_DEV->TXBUF = (uint8_t)out;
-    while (!(SPI_IF & SPI_IE_RX_BIT)) {}
-    tmp = (char)SPI_DEV->RXBUF;
-    if (in) {
-        *in = tmp;
+
+    assert(out || in);
+
+    /* if we only send out data, we do this the fast way... */
+    if (!in) {
+        for (unsigned i = 0; i < length; i++) {
+            while (!(SPI_IF & SPI_IE_TX_BIT)) {}
+            SPI_DEV->TXBUF = (uint8_t)out[i];
+        }
+        while (!(SPI_IF & SPI_IE_RX_BIT)) {}
+        SPI_DEV->RXBUF;
+    }
+    else if (!out) {
+        for (unsigned i = 0; i < length; i++) {
+            SPI_DEV->TXBUF = 0;
+            while (!(SPI_IF & SPI_IE_RX_BIT)) {}
+            in[i] = (char)SPI_DEV->RXBUF;
+        }
     }
-    return 1;
+    else {
+        char tmp;
+        for (unsigned i = 0; i < length; i++) {
+            tmp = (out) ? out[i] : 0;
+            while (!(SPI_IF & SPI_IE_TX_BIT)) {}
+            SPI_DEV->TXBUF = (uint8_t)tmp;
+            while (!(SPI_IF & SPI_IE_RX_BIT)) {}
+            in[i] = (char)SPI_DEV->RXBUF;
+        }
+    }
+
```

Comment:
It leads to the same assembly, but I'd prefer a little more C-like code:


```
unsigned n = length;

while(n--) {
    SPI_DEV->TXBUF = *out++;
}
```

- makes the for loops less verbose
- saves the compiler from doing all the dereferences (```in[i] = x``` ...)